### ISSUE TITLE
Very different results for same dataset during training and using predict function

### ISSUE THREAD

**Issue by ririya** [keras-team/keras#12771]
I have a video classification problem with 2 classes and I'm using a Resnet50 + LSTM architecture. 

During training, the accuracy result of each mini-batch is pretty consistent and close to 100%. However after the network is already trained if I test the training dataset with the predict or predict_generator functions I get 100% accuracy for one of the classes and 0% for the other. 

Does anyone have any idea why this might happen?

**Comment by ririya** [comment]
I'm pretty sure I didn't make any mistakes, I'm even using the same generator function for fit_generator and predict_generator

**Comment by jvishnuvardhan** [comment]
Please fill the issue template [here](https://github.com/keras-team/keras/issues/new/choose). Could you update them if they are relevant in your case, or leave them as N/A? Along with the template, please provide as many details as possible to find the root cause of the issue. It would be great if you can provide a small code to reproduce the error. Thanks!

**Comment by ririya** [comment]
I just found out that one of my videos was giving me trouble and if I got rid of that video for training, the network would behave properly and not even overfit. That same video would not give me trouble if I used it for test instead of training. So I started to think that was something wrong with my data.

I realized that I was normalizing my images by dividing by 2^16 since my input was uint16. However my images are dark and most of the pixels are less than 1000. I fixed it by dividing by 5000 instead of 2^16 and the network started to behave properly. Maybe that one video had smaller values and that caused some instability due to very small values after dividing by 2^16.

This seems to be fixed now but still a very interesting behavior.

**Comment by jvishnuvardhan** [comment]
I am closing the issue as you mentioned the issue was fixed. If I am mistaken, then please resubmit your issue using a template from [here](https://github.com/tensorflow/tensorflow/issues/new/choose). We ask users to use the template because it reduces overall time to resolve a new issue by avoiding extra communication to get to the root of the issue. We will close this issue in lieu of the new one you will create from the template. Thank you for your cooperation.

**Comment by ririya** [comment]
I started having issues again. I still don't know the exact cause of the problem... My training accuracy from the metric function is around 99% (I created a custom function to give the mean of the diagonal of the confusion matrix) , but using the predict function I get 80% / 20% for one of the classes.

`def single_class_accuracy(class_id):
    def fn(y_true, y_pred):
        class_id_true = K.argmax(y_true, axis=-1)
        class_id_preds = K.argmax(y_pred, axis=-1)
        accuracy_mask = K.cast(K.equal(class_id_true, class_id), K.floatx())
        class_acc_tensor = K.cast(K.equal(class_id_true, class_id_preds), K.floatx()) * accuracy_mask
        class_acc = K.sum(class_acc_tensor) / K.maximum(K.sum(accuracy_mask), 1)
        return class_acc
    return fn

def conf_matrix_diag_acc(numClasses, batch_size):
    def fn_conf_matrix_diag_acc(y_true, y_pred):

        val = np.zeros((batch_size,numClasses))
        class_acc_total = K.variable(value=val, dtype='float32')

        for c in range(numClasses):

            curr_fn = single_class_accuracy(c)
            class_acc = curr_fn(y_true,y_pred)
            class_acc_total = class_acc_total + class_acc

        return class_acc_total/numClasses
    return fn_conf_matrix_diag_acc`
